From 8c6c3fb0bceba87045eccadcfa50129ea95a6ebf Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Wed, 14 Nov 2018 01:36:11 +0100
Subject: [PATCH] hurd: Fix build with GCC 9

	* sysdeps/mach/hurd/dl-sysdep.c (check_no_hidden): Use
	__attribute_copy__ to copy attributes from name. Drop static qualifier
	to avoid warnings about leaf attribute not having effect on static
	functions.
---
 ChangeLog                     | 7 +++++++
 sysdeps/mach/hurd/dl-sysdep.c | 5 +++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 837c167a0b..149f991b70 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2018-11-14  Samuel Thibault  <samuel.thibault@ens-lyon.org>
+
+	* sysdeps/mach/hurd/dl-sysdep.c (check_no_hidden): Use
+	__attribute_copy__ to copy attributes from name. Drop static qualifier
+	to avoid warnings about leaf attribute not having effect on static
+	functions.
+
 2018-11-13  Florian Weimer  <fweimer@redhat.com>
 
 	* malloc/malloc.c (fastbin_push_entry): New function.
diff --git a/sysdeps/mach/hurd/dl-sysdep.c b/sysdeps/mach/hurd/dl-sysdep.c
index b72913d3ba..f53abe8de1 100644
--- a/sysdeps/mach/hurd/dl-sysdep.c
+++ b/sysdeps/mach/hurd/dl-sysdep.c
@@ -283,8 +283,9 @@ _dl_sysdep_start_cleanup (void)
 /* This macro checks that the function does not get renamed to be hidden: we do
    need these to be overridable by libc's.  */
 #define check_no_hidden(name)				\
-  static __typeof (name) __check_##name##_no_hidden	\
-       __attribute__ ((alias (#name)));
+  __typeof (name) __check_##name##_no_hidden		\
+       __attribute__ ((alias (#name)))			\
+       __attribute_copy__ (name);
 
 /* Open FILE_NAME and return a Hurd I/O for it in *PORT, or return an
    error.  If STAT is non-zero, stat the file into that stat buffer.  */
-- 
2.30.0

